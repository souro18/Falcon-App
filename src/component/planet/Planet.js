import React ,{useRef, useImperativeHandle ,forwardRef} from 'react'

const Planet = (props,ref) => {
    const selectRef = useRef(null);
    useImperativeHandle(ref, () => ({
        load: () => {
            selectRef.current.focus();
            selectRef.current.value = 'default'
        }
      }));
    return (
        <div className= "planets form-group">
            <select onChange={ (e) => props.onInput(e.target.value)} defaultValue='default' ref={selectRef}>
                <option disabled value='default' >Planets</option>
                {props.planet.map( p => 
                    <option value={p.name} key={p.name} data-testid="planets">{p.name} {p.distance}</option>
                    )}
            </select>
        </div>
    )
}
export default forwardRef(Planet)