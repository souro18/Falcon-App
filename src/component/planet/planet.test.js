import React from 'react';
import {render , cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'

import Planet from './Planet'

afterEach(cleanup)
const planets = [
    {
        name: "abs",
        distance : 200
    },
    {
        name: "earth",
        distance: 300
    }
]
it("renders planet" , ()=> {
    const {asFragment} = render(<Planet planet={planets} onInput={ e => {}}/>)
    expect(asFragment()).toMatchSnapshot()
})
it("renders correct number of planets" , ()=> {
    const {getAllByTestId } = render(<Planet planet={planets} onInput={ e => {}}/>)
    expect(getAllByTestId('planets').length).toBe(planets.length)
})