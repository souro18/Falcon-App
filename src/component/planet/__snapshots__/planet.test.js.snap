// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders planet 1`] = `
<DocumentFragment>
  <div
    class="planets form-group"
  >
    <select>
      <option
        disabled=""
        selected=""
        value="default"
      >
        Planets
      </option>
      <option
        data-testid="planets"
        value="abs"
      >
        abs 200
      </option>
      <option
        data-testid="planets"
        value="earth"
      >
        earth 300
      </option>
    </select>
  </div>
</DocumentFragment>
`;
